chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -338,6 +344,59 @@ impl PeerManagerActor {
                         }
                     }
                 });
+                // Periodically process pending Tier3 requests.
+                arbiter.spawn({
+                    let clock = clock.clone();
+                    let state = state.clone();
+                    let arbiter = arbiter.clone();
+                    let mut interval = time::Interval::new(clock.now(), PROCESS_TIER3_REQUESTS_INTERVAL);
+                    async move {
+                        loop {
+                            interval.tick(&clock).await;
+
+                            if let Some(request) = state.tier3_requests.lock().pop_front() {
+                                arbiter.spawn({
+                                    let clock = clock.clone();
+                                    let state = state.clone();
+                                    async move {
+                                        if let Some(response) = match request.body {
+                                            Tier3RequestBody::StatePartRequest(shard_id, sync_hash, part_id) => {
+                                                state
+                                                    .client
+                                                    .send_async(StateRequestPart { shard_id, sync_hash, part_id })
+                                                    .await
+                                                    .ok()
```

Comment:
I wonder if it would be more helpful when logging below when the result is `None` to not call `.ok()` on `send_async()` here and instead just log that error separately so we can see which one of the `AsyncSendError` variants it is? Otherwise if we log this error it will be harder to tell what happened